packages/core/src/shared/sam/debugger/goSamDebug.ts

```diff
@@ -187,7 +189,9 @@ async function makeInstallScript(debuggerPath: string, isWindows: boolean): Prom
 
     // It's fine if we can't get the latest Delve version, the Toolkit will use the last built one instead
     try {
-        const goPath: string = JSON.parse(execFileSync('go', ['env', '-json']).toString()).GOPATH
+        const goPath: string = JSON.parse(
+            (await ChildProcess.exec('go', ['env', '-json'], { collect: true })).stdout
+        ).GOPATH
```

Comment:
Use a local var is a bit more terse and readable, at the cost of having to claim a name.
But generally, invoking a process almost always should involve checking an exit code (though that's out of scope for this PR), so inlining `exec()` or `run()` is unusual and likely discouraged. .

```suggestion
        const r = await ChildProcess.exec('go', ['env', '-json'], { collect: true })
        const goPath: string = JSON.parse(r.stdout).GOPATH
```

Reply:
Ideally checking the exit code would be part of some higher-level API provided by ChildProcess right?

Reply:
> Ideally checking the exit code would be part of some higher-level API provided by ChildProcess right?

Failure handling in some way needs to be done by the caller, typically